### ISSUE TITLE
Poll not found in Administration and mail invitaions > error 404

### ISSUE THREAD

**Issue by Dennis1993** [nextcloud/polls#2253]
## What is going wrong? What did you observe?
If I open a poll from another user the page loads with error 404 not found.
Some users tell me if they are create a poll and send a mail to the users the poll is not opened. Error 404 ist shown with poll not found
Link is: /index.php/apps/polls/not-found

**Steps to reproduce the behavior**
1. Go to Polls
2. Click on Administration
3. Open a poll from another user

See this example:
![Animation](https://user-images.githubusercontent.com/1473674/153643099-c29ebb00-09a8-48a2-a059-32ebcdc9f6e3.gif)

## How should it work?
The poll opens

## Information about your polls installation
**Polls version:** version 3.4.2 and now updated to 3.5.2 but the problem still exists

**Fresh installation or update from a prior version (from which one)?**
- [ ] First time install
- [X] Update from a prior version
  Version number of the previous version: 3.4.2

**How did you install this version?(Appstore or describe installation)**
- [X] Installed from the appstore
- [ ] Installed via occ
- [ ] Installed via extracting downloaded package to the apps folder

## Information about your Instance of Nextcloud
**Nextcloud version:** (see Nextcloud admin page)
22.2.3

Enabled:
  - activity: 2.15.0
  - bbb: 2.2.0
  - bruteforcesettings: 2.3.0
  - calendar: 3.0.5
  - circles: 22.1.1
  - cloud_federation_api: 1.5.0
  - comments: 1.12.0
  - contactsinteraction: 1.3.0
  - dav: 1.19.0
  - encryption: 2.10.0
  - federatedfilesharing: 1.12.0
  - federation: 1.12.0
  - files: 1.17.0
  - files_rightclick: 1.1.0
  - files_sharing: 1.14.0
  - files_trashbin: 1.12.0
  - files_versions: 1.15.0
  - files_videoplayer: 1.11.0
  - firstrunwizard: 2.11.0
  - logreader: 2.7.0
  - lookup_server_connector: 1.10.0
  - nextcloud_announcements: 1.11.0
  - notifications: 2.10.1
  - oauth2: 1.10.0
  - photos: 1.4.0
  - polls: 3.5.2
  - privacy: 1.6.0
  - provisioning_api: 1.12.0
  - ransomware_protection: 1.12.0
  - serverinfo: 1.12.0
  - settings: 1.4.0
  - sharebymail: 1.12.0
  - systemtags: 1.12.0
  - theming: 1.13.0
  - twofactor_backupcodes: 1.11.0
  - updatenotification: 1.12.0
  - user_ldap: 1.12.1
  - viewer: 1.6.0
  - weather_status: 1.2.0
  - workflowengine: 2.4.0
Disabled:
  - accessibility
  - admin_audit
  - dashboard
  - files_external
  - files_pdfviewer
  - password_policy
  - recommendations
  - support
  - survey_client
  - text
  - user_status

<summary>Nextcloud Configuration</summary>
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "cloud.my-domain.de"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwritewebroot": "\/",
        "overwriteprotocol": "https",
        "dbtype": "mysql",
        "version": "22.2.3.0",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "default_phone_region": "DE",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "activity_expire_days": 30,
        "loglevel": 2,
        "ldapIgnoreNamingRules": false,
        "forcessl": true,
        "theme": "",
        "lost_password_link": "disabled",
        "auth.webauthn.enabled": false,
        "maintenance": false,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "trashbin_retention_obligation": "auto,7",
        "ldapProviderFactory": "\\OCA\\User_LDAP\\LDAPProviderFactory",
        "overwritehost": "cloud.my-domain.de",
        "overwrite.cli.url": "https:\/\/cloud.my-domain.de",
        "skeletondirectory": "",
        "force_language": "de_DE",
        "force_locale": "de_DE",
        "allow_user_to_change_display_name": false,
        "updater.release.channel": "stable",
        "encryption.key_storage_migrated": false,
        "app_install_overwrite": [
            "bruteforcesettings"
        ],
        "mail_sendmailmode": "pipe",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "25",
        "mail_smtpauth": 1,
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***"
    }
}

### Server configuration (decide, if you think it is helpful)

**Database:**
- [ ] MySql version:
- [X] MariaDB version: 10.3
- [ ] PostgreSQL version:
- [ ] Oracle version:
- [ ] SQLite:
- [ ] Other (add name and version):

**PHP version:**
- [ ] 7.3 or lower
- [ ] 7.4
- [X] 8.0

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
Yes, all users are LDAP accounts. Only the admin account is a local account

### Client configuration
**Device:** 
- [X] PC
- [ ] Mac
- [ ] Mobile phone
- [ ] Tablet

**Browser:** 
- [ ] Firefox version:
- [ ] Chrome version:
- [ ] Safari version:
- [X] Microsoft Edge: 98

## Logs

<details>
<summary>Nextcloud log (data/nextcloud.log)</summary>
empty

</details>

<details>
<summary>Browser console log</summary>
![console](https://user-images.githubusercontent.com/1473674/153644977-3a37cac6-ab7b-4769-bc33-21df5edecc84.png)

</details>

**Comment by dartcafe** [comment]
In this case you have no access to the poll, because the user created a hidden poll and did not invite you. This is by intention, because the owner of the poll did not give you access to this (private) poll. 

But: Nevertheless, the bug is that you can click the entry. The administration page give administrators the possibility to delete polls or take them over, if other access isn't possible anymore (i.e. because the owner is deleted from the instance).

**Comment by dartcafe** [comment]
You state, that the error also occurs, if the poll ist sent via email. Was the mail sent via invitation or did the user send a mail with his email client, pasting the address into the mail? 

In the first case, I need more information about the sent share, because I cannot reproduce that. 

In the second case, please check the access of the poll in the shares tab.

**Comment by Dennis1993** [comment]
The first case.
I test it tomorrow and post an animation here.

maybe we should „permission denied“ instead of „404 not found“? 🤔

thanks for your quick response 👍

**Comment by Dennis1993** [comment]
Hey,

I have tested it and found the problem. Some users told me that some users can't vote for a poll by an mail share. I tried to reproduce this.

The problem is (all users are LDAP accounts):
- **User A** created a poll
- **User A** shared the poll with **User B**, **User C**, **User D**...
- Now he typed in **user-h@example.com** and send the invitation. But the **User H** has an account!
- The **User H** see the mail and clicked on the link in it. The User H is logged in to NextCloud and the error appears
- If the **User H** logged out and use the link again it works

So the problem is that the accounts are not linked. We should check: If user mail is stored in NextCloud, then link it to the user account. If not, then the external flow is used.

I hope you can understand that :)

Thanks

**Comment by dartcafe** [comment]
Yes I understand, but the problem is that the poll owner created a public link, which is not linked to a user of the site. Email shares are links which are only intended to be used as a publicv share. 

We can check if we can manage to link an email address to an existing user. Would be good to have a separate [feature request](https://github.com/nextcloud/polls/issues/new?assignees=&labels=&template=feature_request.md) for that. 

This issue could be closed then.
